core/templates/dev/head/domain/collection/GuestCollectionProgressService.js

```diff
@@ -0,0 +1,146 @@
+// Copyright 2017 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service that records progress guests make during a collection
+ * playthrough. Note that this service does not currently support saving a
+ * user's progress when they create an account.
+ */
+
+// TODO(bhenning): Move this to a shared service which stores state across the
```

Comment:
Could you explain a bit more about what "shared" means in this context? (Isn't this already a singleton service?)

Reply:
Thanks for pointing this out; 'shared' was attributed to the wrong noun. What I meant is we should have a service that stores state which is shared across the frontend. I also provided a few examples for context. This is a lead-in for the saving exploration progress work.